test/hosting/test_HostingBasic.cs

```diff
@@ -139,5 +139,37 @@ public static void TestCommandFromNative()
                 File.Delete(target);
             }
         }
+
+        [Fact]
+        public static void TestAddType()
+        {
+            string code = @"
+                using System;
+                public class Foo
+                {
+                    public Foo(string name, string path)
+                    {
+                        this.Name = name;
+                        this.Path = path;
+                    }
+
+                    public string Name;
+                    public string Path;
+                }
+            ";
+
+            using (System.Management.Automation.PowerShell ps = System.Management.Automation.PowerShell.Create())
+            {
+                ps.AddCommand("Add-Type").AddParameter("TypeDefinition", code).Invoke();
+                ps.Commands.Clear();
+
+                var results = ps.AddScript("[Foo]::new('Joe', 'Unknown')").Invoke();
```

Comment:
Should we be more strong in tests?
```suggestion
                Collection<PSObject> results = ps.AddScript("[Foo]::new('Joe', 'Unknown')").Invoke();
```

Reply:
I think it's fine to use `var` here. The purpose of the tests here is to make sure basic things work in hosting scenario, not specifically testing the API itself.
Also, changing signature of the API is too much a breaking change that will probably never be accepted :)